### PR TITLE
Doc/doxy update

### PR DESCRIPTION
**Pull request by adityaghantasala**
While recently generating doxygen documentation for a friend, I recognized that the current doxygen file can use some updates and options. This PR introduces a more detailed version of the doxygen file and more detailed options. 

Also when extracting the documentation from Python files, I found that "doxypypy"  
`https://github.com/Feneric/doxypypy`  did a better job when combined with doxygen. So I added a `README` to folder to point to this installation procedure. 

Please feel free to add / change options in the doxygen file.

### CODE REVIEW

**Comment by philbucher** [comment]
wait doxygen compiles? :sweat_smile: 
I though it was broken

**Comment by adityaghantasala** [comment]
What do you mean doxygen is broken ?

**Comment by philbucher** [comment]
well in the almost 5 years I am using Kratos now I never saw anyone using it
But good to know it works :)

File: documents/kratos_doxyfile [hunk 1]
```diff
@@ -1,90 +1,122 @@
...
+
+PROJECT_BRIEF          = "KRATOS Multiphysics ("Kratos") is a framework for building parallel, multi-disciplinary simulation software, aiming at modularity, extensibility, and high performance. Kratos is written in C++, and counts with an extensive Python interface."
+
+# With the PROJECT_LOGO tag one can specify a logo or an icon that is included
+# in the documentation. The maximum height of the logo should not exceed 55
+# pixels and the maximum width should not exceed 200 pixels. Doxygen will copy
+# the logo to the output directory.
 
-PROJECT_NUMBER         = 2.1.0
+PROJECT_LOGO           =
```
Comment (roigcarlo) [inline documents/kratos_doxyfile]: We should probably fill this as well

File: documents/kratos_doxyfile [hunk 2]
```diff
@@ -1,90 +1,122 @@
...
+# If the CREATE_SUBDIRS tag is set to YES then doxygen will create 4096 sub-
+# directories (in 2 levels) under the output directory of each output format and
+# will distribute the generated files over these directories. Enabling this
+# option can be useful when feeding doxygen a huge amount of source files, where
+# putting all generated files in the same directory would otherwise causes
+# performance problems for the file system.
+# The default value is: NO.
 
-CREATE_SUBDIRS         = YES
+CREATE_SUBDIRS         = NO
```
Comment (roigcarlo) [inline documents/kratos_doxyfile]: We should set this to `YES` for Kratos, otherwise it become a mess

File: documents/kratos_doxyfile [hunk 3]
```diff
@@ -99,1512 +131,2337 @@ ABBREVIATE_BRIEF       = "The $name class" \
...
-# file names in lower-case letters. If set to YES upper-case letters are also
+# If the CASE_SENSE_NAMES tag is set to NO then doxygen will only generate file
+# names in lower-case letters. If set to YES, upper-case letters are also
 # allowed. This is useful if you have classes or files whose names only differ
 # in case and if your file system supports case sensitive file names. Windows
 # and Mac users are advised to set this option to NO.
+# The default value is: system dependent.
 
-CASE_SENSE_NAMES       = NO
+CASE_SENSE_NAMES       = YES
```
Comment (roigcarlo) [inline documents/kratos_doxyfile]: That one must be `NO`.

File: documents/kratos_doxyfile [hunk 4]
```diff
@@ -99,1512 +131,2337 @@ ABBREVIATE_BRIEF       = "The $name class" \
...
-
-SHOW_DIRECTORIES       = NO
-
-# Set the SHOW_FILES tag to NO to disable the generation of the Files page.
-# This will remove the Files entry from the Quick Index and from the
-# Folder Tree View (if specified). The default is YES.
+# Set the SHOW_FILES tag to NO to disable the generation of the Files page. This
+# will remove the Files entry from the Quick Index and from the Folder Tree View
+# (if specified).
+# The default value is: YES.
```
Comment (roigcarlo) [inline documents/kratos_doxyfile:675]: Has the `SHOW_DIRECTORIES` option been deprecated?

File: documents/kratos_doxyfile [hunk 5]
```diff
@@ -99,1512 +131,2337 @@ ABBREVIATE_BRIEF       = "The $name class" \
...
 #---------------------------------------------------------------------------
 
-# If the GENERATE_HTML tag is set to YES (the default) Doxygen will
-# generate HTML output.
+# If the GENERATE_HTML tag is set to YES, doxygen will generate HTML output
+# The default value is: YES.
 
 GENERATE_HTML          = YES
 
-# The HTML_OUTPU
```
Comment (roigcarlo) [inline documents/kratos_doxyfile:1514]: Mostly a reminder to myself: I think that could be usefull, but lets see how it works without it.

Comment (roigcarlo) [inline documents/kratos_doxyfile:1647]: I think we should set this to `NO`.

**Review by roigcarlo — CHANGES_REQUESTED** [review]
Just a couple of comments about some flags. Rest looks fine!

I am not aware of what most of the new options do, so I trust you on those.

File: documents/kratos_doxyfile (hunk 5 above)
Comment (roigcarlo) [reply documents/kratos_doxyfile:1514]: That one should be pointing to the mathjs thing.. anyway is not important to change it now.

File: documents/kratos_doxyfile (hunk 2 above)
Comment (adityaghantasala) [reply documents/kratos_doxyfile]: Ok

File: documents/kratos_doxyfile (hunk 3 above)
Comment (adityaghantasala) [reply documents/kratos_doxyfile]: OK

File: documents/kratos_doxyfile (hunk 5 above)
Comment (adityaghantasala) [reply documents/kratos_doxyfile:1647]: But if we put it to NO we do not get any documentation to view on a browser no ?
We will only get a latex version. !

File: documents/kratos_doxyfile (hunk 4 above)
Comment (adityaghantasala) [reply documents/kratos_doxyfile:675]: I thinks some how they are merged. 
This is how I see the files now 

image


So files are grouped into folders ... so show_directories is some how switched on !

File: documents/kratos_doxyfile (hunk 5 above)
Comment (roigcarlo) [reply documents/kratos_doxyfile:1647]: Is exactly as you say. I think that's a github problem :S. Actually I was marking the line that sets `GENERATE_LATEX         = YES` so it becomes `NO` to get only the html. But it displays another thing in the review page:
image

Comment (roigcarlo) [reply documents/kratos_doxyfile:1647]: Anyway its fine, as long as the html is generated I don't really care about latex. Was just to clean up. We can merge

**Review by roigcarlo — DISMISSED** [review]

File: documents/kratos_doxyfile (hunk 5 above)
Comment (adityaghantasala) [reply documents/kratos_doxyfile:1647]: Ok.

**Comment by adityaghantasala** [comment]
I am testing with the USE_MDFILE_AS_MAINPAGE so to use the README.md as the main page including the logo. One I get it running I will push that and we can merge !

**Comment by adityaghantasala** [comment]
@roigcarlo   Now the main page looks like this 

image

**Review by roigcarlo — APPROVED** [review]
👍 👍

**Comment by adityaghantasala** [comment]
Thank you  !

**Comment by philbucher** [comment]
@adityaghantasala it would be very helpful if you could add to the readme how to create the doxygen files. (esp for ppl like me that never used doxygen :) )
Thanks in advance

**Comment by loumalouomega** [comment]
@roigcarlo is time to retake this: https://github.com/mosra/m.css

**Pull request merged**